internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -348,14 +348,6 @@ func resourceOrchestratedVirtualMachineScaleSetCreate(d *pluginsdk.ResourceData,
 		log.Printf("[DEBUG] Orchestrated Virtual Machine Scale Set %q (Resource Group %q) has a Health Extension defined", id.Name, id.ResourceGroup)
 	}
 
-	if v, ok := d.Get("extension_operations_enabled").(bool); ok {
-		_, extensionsExists := d.GetOk("extension")
-		if v && !extensionsExists {
-			return fmt.Errorf("%q can not be set to %q if the %q field does not contain any extensions", "extension_operations_enabled", "true", "extension")
-		}
-		virtualMachineProfile.OsProfile.AllowExtensionOperations = utils.Bool(v)
```

Comment:
virtualMachineProfile.OsProfile is nil at this point, have moved this block below